### PR TITLE
applayer: add test for bug 7044

### PR DESCRIPTION
**Pull request by inashivb**
## Ticket

If your pull request is related to a Suricata ticket, please provide
the full URL to the ticket here so this pull request can monitor
changes to the ticket status:

Redmine ticket: https://redmine.openinfosecfoundation.org/issues/7044

Note: This shows truncate fn is not needed. It is already taken care of.

### CODE REVIEW

**Comment by inashivb** [comment]
@catenacyber what do you think of this test?

File: tests/truncate-applayer-test/test.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+args:
+  - -k none --set stream.reassembly.depth=8kb
+
+checks:
+- filter:
+    count: 1
+    match:
+      event_type: alert
+      alert.signature_id: 1
```
Comment (catenacyber) [inline tests/truncate-applayer-test/test.yaml:9]: Could we check pcap_cnt ?
That this does not happen on flow timeout

Comment (catenacyber) [reply tests/truncate-applayer-test/test.yaml:9]: Also check for dcerpc metadata now that they are available in master  cf https://github.com/OISF/suricata/pull/11151#issuecomment-2182390221

Comment (catenacyber) [reply tests/truncate-applayer-test/test.yaml:9]: Oh and more important, we should test for a dcerpc event, not an alert !!

File: tests/truncate-applayer-test/test.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+- filter:
+    count: 0    # because of the depth constraints
```
Comment (catenacyber) [inline tests/truncate-applayer-test/test.yaml:11]: We could also have a second test without the `stream.reassembly.depth=8kb` to show that we get the alert

File: tests/truncate-applayer-test/test.yaml (hunk 1 above)
Comment (inashivb) [reply tests/truncate-applayer-test/test.yaml:9]: The alert happens at an exact dcerpc event. I had modified the stub_size settings in rule to have the exact buffer size where stream depth changes start to take place.
I will also add the dcerpc event though as it makes it easier to correlate the alert then. :)
Is there another reason that I'm missing perhaps for which you want to see the dcerpc event?

Comment (catenacyber) [reply tests/truncate-applayer-test/test.yaml:9]: dcerpc event logging and alert are 2 different code paths.
We should test dcerpc event logging (as it is more bidirectional than alert)
cf https://github.com/OISF/suricata/pull/11151#issuecomment-2182390221

Comment (inashivb) [reply tests/truncate-applayer-test/test.yaml:9]: ah I had missed the part explaining logger path in your comment. 🤦🏽‍♀️ 
Thank you! Taking one more look around the code paths and truncate usages.

**Pull request closed without merging**